src/dfx/src/lib/deps.rs

```diff
@@ -0,0 +1,239 @@
+use crate::lib::{environment::Environment, error::DfxResult};
+use dfx_core::{
+    config::cache::get_cache_root,
+    fs::composite::ensure_parent_dir_exists,
+    json::{load_json_file, save_json_file},
+};
+
+use std::{
+    collections::BTreeMap,
+    path::{Path, PathBuf},
+};
+
+use anyhow::bail;
+use candid::Principal;
+use fn_error_context::context;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct PulledJson {
+    pub named: BTreeMap<String, Principal>,
+    pub canisters: BTreeMap<Principal, PulledCanister>,
+}
```

Comment:
Also: what is the rationale for/purpose of the `named` field in pulled.json?

Reply:
I refactored pulled.json by removing `named` field` and add `name: Option<String>` to each `PulledCanister`.
The name will be displayed alongside the canister_id in `dfx deps init` and `dfx deps deploy`, e.g. "yhgn4-myaaa-aaaaa-aabta-cai (dep_b)"